server/clip_server/executors/clip_torch.py

```diff
@@ -32,49 +35,70 @@ def __init__(
         else:
             self._pool = Pool(processes=num_worker_preprocess)
 
+        self.summary_text = self.get_summary('text_preproc_second', 'Time to preprocess text')
+        self.summary_image = self.get_summary('image_preproc_second', 'Time to preprocess image')
+        self.summary_encode = self.get_summary('encode_second', 'Time to encode')
+    def get_summary(self, title, details):
```

Comment:
There are already some base summary initialize in the BaseExecutor, but here proces text and proces image are specific so this executor so they would always need to be define here
I note for the Tensorboard for custom tracking name.

Reply:
i agree with Felix on this one, I'd prefer:

**Best**: No change to the executor at all. Injection happens inside the core. Think about Hub Executors, how can we change all of them, especially those are not owned by us.
**Okay**: injection happens at `@requests(..., monitor=True, summary=True)`
**Bad**: need a lot of lines of change like the current PR. This is unacceptable to me, also to all of our Hub Executors.

Reply:
I might be unclear.

The core will support all bunch of metrics which will be common to all Executor. Like the latency on gateway,head,worker side and the time that the function wrapper by requests takes. 
This will be common to all Executor and not a single line of code would be needed on the Executor side.

In addition of that I want to have an extra feature that allow Executor developer to customize and enrich this monitoring by defining by themself new metrics to expose. This is highly relevant for clip as a service as I would like to know how much time is actually used on gpu computation and how much time is used in the text/image preprocessing. The idea with that extra feature is that you can monitoring each sub function that is called inside your request method. And those things can't be abstracted away as it is not common to all executor.